### ISSUE TITLE
Support for groupBy

### ISSUE THREAD

**Issue by ehdrms785** [toss/es-toolkit#399]
Hello, @raon0211 

It would be great if `groupBy` could be added to` es-toolkit`.

Could I try implementing this ?

Ex ) https://lodash.com/docs/4.17.15#groupBy

**Comment by raon0211** [comment]
We actually already have groupBy in our library. Is this insufficient?

https://es-toolkit.slash.page/reference/array/groupBy.html#groupby

**Comment by Vipul-Vermaa** [comment]
How to setup this project on windows?
I am getting error at multiple stage

**Comment by ehdrms785** [comment]
> We actually already have groupBy in our library. Is this insufficient?
> 
> https://es-toolkit.slash.page/reference/array/groupBy.html#groupby

Apologies, it was my mistake. 
I’m not sure why the groupBy method didn’t catch my eye when I was looking for it.
I thought there might be some method for performance improvement,
So I tried using a pure object. 

Although I obtained some meaningful results, using a pure object—thus not utilizing the prototype—seems like a risky approach.

 I’ll definitely participate again as a contributor if I have the opportunity in the future!

<img width="528" alt="image" src="https://github.com/user-attachments/assets/f311ab4c-5dc9-43c7-bf50-643bd33d8a7c">
